stdlib/lang-parser/parserModule/parser.bal

```diff
@@ -0,0 +1,1260 @@
+        } else if (self.LAToken(1) == NUMBER) {
+            self.priorOperator = false;
+            if (self.expOperand == false) {
+                Token invalidToken = self.deleteToken();
+                return true;
+            } else {
+                Token number = self.matchToken(NUMBER, EXPRESSION_NODE);
+                IntegerLiteralNode intLit = {
+                    nodeKind: INTEGER_LITERAL,
+                    tokenList: [number],
+                    number: number.text
+                };
+                SimpleLiteral sLit = intLit;
+                self.expStack.push(sLit);
+                self.expOperand = false;
+                return true;
+            }
+        } else if (self.LAToken(1) == IDENTIFIER) {
+            self.priorOperator = false;
+            if (self.expOperand == false) {
+                Token invalidToken = self.deleteToken();
+                return true;
+            } else {
+                Token identifier = self.matchToken(IDENTIFIER, EXPRESSION_NODE);
+                VarRefIdentifier varRef = {
+                    nodeKind: VAR_REF_NODE,
+                    tokenList: [identifier],
+                    varIdentifier: identifier.text
+                };
+                self.expStack.push(varRef);
+                self.expOperand = false;
+                return true;
+            }
+        } else if ((self.LAToken(1) >= ADD && self.LAToken(1) <= UNTAINT) || self.LAToken(1) == COMMA) {            //binary, unary operators and comma
+
+            //if the expression stack is empty then that means no prior expression , so this is a unary expr
+            if (self.expStack.isEmpty() == true || self.priorOperator == true) {
+                if (self.LAToken(1) == SUB || self.LAToken(1) == ADD || (self.LAToken(1) >= 25 && self.LAToken(1) <= 27)) {                    // unary operators
+
+                    if (self.LAToken(1) == SUB) {
+                        Token unarySub = self.matchToken(SUB, EXPRESSION_NODE);
+                        unarySub.tokenType = UNARY_MINUS;
+                        self.oprStack.push(unarySub);
+                        self.expOperand = true;
+                        self.priorOperator = true;
+                        return true;
+                    } else if (self.LAToken(1) == ADD) {
+                        Token unaryAdd = self.matchToken(ADD, EXPRESSION_NODE);
+                        unaryAdd.tokenType = UNARY_PLUS;
+                        self.oprStack.push(unaryAdd);
+                        self.expOperand = true;
+                        self.priorOperator = true;
+                        return true;
+                    } else {
+                        Token unaryOpToken = self.matchToken(self.LAToken(1), EXPRESSION_NODE);
+                        self.oprStack.push(unaryOpToken);
+                        self.expOperand = true;
+                        self.priorOperator = true;
+                        return true;
+                    }
```

Comment:
L904-L906 seem to be repeated for all if/else blocks here? Shall we move them out to immediately after the else block?